### PR TITLE
Clean up packet_dropped

### LINKED ISSUE #25: Additional triggers and info for dropped packets
The current text lists about 8 reasons for dropping packets.
Microsoft's implementation lists 60+ individual reasons (via @nibanks)

Some of those:

> LogDrop("Retry sent to server");
LogDrop("Already received server response");
LogDrop("No room for ODCID");
LogDrop("Invalid ODCID");
LogDrop("InitialToken alloc failed");
LogDrop("OrigCID alloc failed");
LogDrop("Max deferred datagram count reached");
LogDrop("Key no longer accepted");
LogDrop("SH packet during version negotiation");
LogDrop("Too short for HP");
LogDrop("Packet number too big");
LogDrop("Payload length less than encryption tag");
LogDrop("Generate new packet keys");
LogDrop("Decryption failure");
LogDrop("Invalid SH Reserved bits values");
LogDrop("Invalid LH Reserved bits values");
LogDrop("Duplicate packet number");
LogDrop("Key no longer accepted (batch)");
LogDrop("Failed to compute HP mask");
LogDrop("Different remote address");
LogDrop("Too small for Packet->Invariant");
LogDrop("LH no room for DestCID");
LogDrop("Zero length DestCID");
LogDrop("LH no room for SourceCID");
LogDrop("SH no room for DestCID");
LogDrop("DestCID don't match");
LogDrop("SourceCID don't match");
LogDrop("Greater than allowed max CID length");
LogDropWithValue("Invalid client/server packet type", Packet->LH->Type);
LogDrop("Invalid LH FixedBit bits values");
LogDrop("Long header has invalid token length");
LogDropWithValue("Long header has token length larger than buffer length", TokenLengthVarInt);
LogDrop("Long header has invalid payload length");
LogDropWithValue("Long header has length larger than buffer length", LengthVarInt);
LogDropWithValue("Long Header doesn't have enough room for packet number",
LogDrop("Invalid SH FixedBit bits values");
LogDrop("Non-initial packet not matched with a Connection");
LogDrop("Retry received after initial");

I feel that we don't need to list things in this level of detail in the qlog spec (the "trigger" field allows any text anyway). However, maybe some guidance text on this would be helpful and maybe a few more suggested triggers would be interesting.

### PR DESCRIPTION
**Pull request by rmarx**
Fixes #25.

### CODE REVIEW

File: draft-ietf-quic-qlog-quic-events.md [hunk 1]
```diff
@@ -778,42 +778,47 @@ when keeping track of packet coalescing.
...
 {: #transport-packetdropped-def title="TransportPacketDropped definition"}
 
-Note: sometimes packets are dropped before they can be associated with a
-particular connection (e.g., in case of "unsupported_version"). This situation is
-discussed more in {{handling-unknown-connections}}.
+Some example situations for each of the trigger categories include:
 
-Note: for more details on "datagram_id", see {{transport-datagramssent}}. It is only needed
-when keeping track of packet coalescing.
+- internal_error: not initialize, out of memory
```
Comment (LPardue) [inline draft-ietf-quic-qlog-quic-events.md:813]:
```suggestion
- internal_error: not initialized, out of memory
```

File: draft-ietf-quic-qlog-quic-events.md [hunk 2]
```diff
@@ -778,42 +778,47 @@ when keeping track of packet coalescing.
...
+- rejected: limits reached, DDoS protection, unwilling to track more paths, duplicate packet
+- unsupported: unknown or unsupported version. See also {{handling-unknown-connections}}.
+- invalid: packet parsing or validation error
+- connection_unknown: Connection ID is not known
```
Comment (LPardue) [inline draft-ietf-quic-qlog-quic-events.md:817]: more generic version 

```suggestion
- connection_unknown: packet does not relate to a known connection
```

**Review by LPardue — APPROVED** [review]
LGTM modulo nits

**Comment by rmarx** [comment]
Hey @nibanks, we think this PR is ready to be merged. If you have some time this week for a quick review, that would be splendid :)

**Review by nibanks — APPROVED** [review]

**Pull request merged**